### DISCUSSION TITLE
The road ahead for the SOFP book

### DISCUSSION THREAD

**Discussion by winitzki** [winitzki/sofp#30] [Ideas]
What remains to be done for the SOFP book to be ready:

- [x] finish rewriting Appendix D
- [x] finish writing the LJT algorithm, as part of Chapter 5; eliminate Appendix C
- [x] rewrite parts of Chapter 5 to make the presentation clearer. The Curry-Howard correspondence needs to be explained in a shorter logical sequence, removing the detour to Boolean logic.
- [x] write chapter 11 (applicative functors)
- [x] write chapter 12 (traversable functors)
- [ ] write chapter 13 (free type constructions)
- [ ] maybe consider adding some material about lenses, although I feel that it will take me a much longer time than I have.

**Comment by winitzki** [comment]
"finish rewriting Appendix D" is complete.

**Comment by winitzki** [comment]
Chapter 11 is complete. Chapter 12 is complete up to one part of the proof of one statement, which seems to require further parametricity results that I do not currently have proved.

**Comment by winitzki** [comment]
Rewriting Chapter 5 and Appendix C is complete.

**Comment by winitzki** [comment]
In Chapter 12, only one small part of one proof remains to be written.

**Comment by winitzki** [comment]
Chapter 12 is complete.

**Comment by winitzki** [comment]
"Finish rewriting Appendix D" is complete yet again. I discovered serious errors in several theorems, and it took me 3 months to fix everything so that Appendix D is correct. I had to reduce the scope of some proofs. For instance, it was incorrect that relational composition satisfies the lifting law. It is not true that lifting of a pullback relation to a functor is again a pullback relation; this is only true for polynomial functors. Now I no longer define the composition of relations or prove any properties of it. As a result, Appendix D is now 9 pages shorter.

**Comment by dmanikhine** [comment]
Is it possible to add references to selected chapters?

**Reply by winitzki** [reply]
I'm not in favor of adding references at the end of chapters or at the end of the book. I'm not really sure what purpose those references would serve. This book is not a research work but a pedagogical textbook. At the same time, there are no other textbooks with similar material to which I could point for more details. I do have footnotes to papers where appropriate but I don't think it would be better to have those footnotes at the end of a chapter or at the end of the book.

**Reply by dmanikhine** [reply]
It is impossible to cover everything in one book.
 Little things that are obvious to the author are often not obvious to the reader.
Such non-obvious things can be revealed in other sources.
Alternative examples may also be given there.
Sources are very important to me.
Please, if possible, indicate them at the end of the chapters.
Thank you very much in advance.

**Reply by winitzki** [reply]
When you say "sources are important" you mean what kind of "sources"? It's not clear to me what you would like to add at the end of the chapters. I did not write my book from any other sources, I have developed my own approach and motivation.
